src/activity/activity.jsx

```diff
@@ -15,46 +18,111 @@ import Histogram from './components/histogram'
 import appStyles from 'src/styles'
 import styles from './activity.scss'
 
-import histogramUsers from '../../public/mocks/histogram-users.json'
-import histogramFeatures from '../../public/mocks/histogram-features.json'
-
 class DailyActivity extends Component {
   constructor (props) {
     super(props)
     this.updateFacet = this.updateFacet.bind(this)
     this.updateGranularity = this.updateGranularity.bind(this)
-    this.state = this.formatState(props.data)
+    this.state = this.formatState(props)
   }
 
-  formatState (data) {
+  formatState (props) {
     return {
-      granularity: GRANULARITIES.Daily,
-      facet: FACETS.Features
+      granularity:
+        (props.granularity && startCase(props.granularity)) ||
+        GRANULARITIES.Daily,
+      facet: (props.facet && startCase(props.facet)) || FACETS.Features,
+      data: props.data,
+      range: props.range || [new Date(), new Date()]
     }
   }
 
-  getUsers (data) {
-    return histogramUsers // .slice(0, 6)
-      .map(d => [d, _max(d)])
+  parseDate (d) {
+    const date = new Date(d)
+    const day = date.getDate() - 1
+    const month = date.getMonth()
+    const year = date.getFullYear()
+    return {
+      day,
+      month,
+      year,
+      len: monthLength(month, year)
+    }
+  }
+
+  formatData (data, getCount) {
+    const months = 12
+    const { range } = this.state
+    const fromSamp = new Date(range[0]).getTime()
+    const toSamp = new Date(range[1]).getTime()
+    const filteredValues = data
+      .sort((a, b) => a.day - b.day)
+      .filter(d => d && d.day >= fromSamp && d.day < toSamp)
+
+    const max = filteredValues.length
+      ? getCount(_maxBy(filteredValues, getCount))
+      : 0
+
+    return [
+      filteredValues.reduce((result, item) => {
+        const { day, month, year, len } = this.parseDate(item.day)
+        result[year] = !result[year] ? new Array(months) : result[year]
```

Comment:
I'm used to instanciate arrays with `[]` for some obscure reasons I forgot but mainly because it is usually in standard eslint conf (http://eslint.org/docs/rules/no-array-constructor). 
Looking it up, there are various places that explain why `new Array` should not be preferred, ie https://coderwall.com/p/h4xm0w/why-never-use-new-array-in-javascript